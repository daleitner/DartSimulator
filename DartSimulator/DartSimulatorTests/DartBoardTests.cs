using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DartSimulator;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Should;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;

namespace DartSimulatorTests
{
	[TestClass]
	[UseReporter(typeof(DiffReporter))]
	public class DartBoardTests
	{
		[TestMethod]
		public void DartBoardShouldHaveGotAllFields()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields = dartBoard.Fields.Select(x => x.ToString());
			Approvals.VerifyAll(fields, "Field");
		}

		[TestMethod]
		public void GetDoubleBullShouldReturnDoubleBullField()
		{
			var dartBoard = DartBoard.GetInstance();
			var doubleBull = dartBoard.GetDoubleBull();
			Approvals.Verify(doubleBull);
		}

		[TestMethod]
		public void GetSingleBullShouldReturnSingleBullField()
		{
			var dartBoard = DartBoard.GetInstance();
			var singleBull = dartBoard.GetSingleBull();
			Approvals.Verify(singleBull);
		}

		[TestMethod]
		public void GetSingleInFieldShouldReturnSingleInFieldWithRightValue()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields =
				dartBoard.Fields.Where(x => x.Type == FieldEnum.SingleIn).Select(x => dartBoard.GetSingleInField(x.Value));
			Approvals.VerifyAll(fields, "Field");
		}
		[TestMethod]
		public void GetSingleOutFieldShouldReturnSingleOutFieldWithRightValue()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields =
				dartBoard.Fields.Where(x => x.Type == FieldEnum.SingleOut).Select(x => dartBoard.GetSingleOutField(x.Value));
			Approvals.VerifyAll(fields, "Field");
		}
		[TestMethod]
		public void GetDoubleFieldShouldReturnDoubleFieldWithRightValue()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields =
				dartBoard.Fields.Where(x => x.Type == FieldEnum.Double).Select(x => dartBoard.GetDoubleField(x.Value));
			Approvals.VerifyAll(fields, "Field");
		}
		[TestMethod]
		public void GetTripleFieldShouldReturnTripleFieldWithRightValue()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields =
				dartBoard.Fields.Where(x => x.Type == FieldEnum.Triple).Select(x => dartBoard.GetTripleField(x.Value));
			Approvals.VerifyAll(fields, "Field");
		}

		[TestMethod]
		public void VerifyNeighboursOfFields()
		{
			var dartBoard = DartBoard.GetInstance();
			string verifyString = "";
			foreach (var field in dartBoard.Fields)
			{
				verifyString += field.ToString() + "\n";
				foreach (var neighbour in field.Neighbours)
				{
					verifyString += "\t" + neighbour.ToString() + "\n";
				}
			}
			Approvals.Verify(verifyString);
		}
	}
}
