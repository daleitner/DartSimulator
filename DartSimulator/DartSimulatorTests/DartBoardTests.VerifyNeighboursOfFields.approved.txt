0
1
	D1
	T1
	20
	18
1
	T1
	25
	20
	18
D1
	0
	1
	D20
	D18
T1
	1
	1
	T20
	T18
2
	D2
	T2
	15
	17
2
	T2
	25
	15
	17
D2
	0
	2
	D15
	D17
T2
	2
	2
	T15
	T17
3
	D3
	T3
	17
	19
3
	T3
	25
	17
	19
D3
	0
	3
	D17
	D19
T3
	3
	3
	T17
	T19
4
	D4
	T4
	18
	13
4
	T4
	25
	18
	13
D4
	0
	4
	D18
	D13
T4
	4
	4
	T18
	T13
5
	D5
	T5
	12
	20
5
	T5
	25
	12
	20
D5
	0
	5
	D12
	D20
T5
	5
	5
	T12
	T20
6
	D6
	T6
	13
	10
6
	T6
	25
	13
	10
D6
	0
	6
	D13
	D10
T6
	6
	6
	T13
	T10
7
	D7
	T7
	19
	16
7
	T7
	25
	19
	16
D7
	0
	7
	D19
	D16
T7
	7
	7
	T19
	T16
8
	D8
	T8
	16
	11
8
	T8
	25
	16
	11
D8
	0
	8
	D16
	D11
T8
	8
	8
	T16
	T11
9
	D9
	T9
	14
	12
9
	T9
	25
	14
	12
D9
	0
	9
	D14
	D12
T9
	9
	9
	T14
	T12
10
	D10
	T10
	6
	15
10
	T10
	25
	6
	15
D10
	0
	10
	D6
	D15
T10
	10
	10
	T6
	T15
11
	D11
	T11
	8
	14
11
	T11
	25
	8
	14
D11
	0
	11
	D8
	D14
T11
	11
	11
	T8
	T14
12
	D12
	T12
	9
	5
12
	T12
	25
	9
	5
D12
	0
	12
	D9
	D5
T12
	12
	12
	T9
	T5
13
	D13
	T13
	4
	6
13
	T13
	25
	4
	6
D13
	0
	13
	D4
	D6
T13
	13
	13
	T4
	T6
14
	D14
	T14
	11
	9
14
	T14
	25
	11
	9
D14
	0
	14
	D11
	D9
T14
	14
	14
	T11
	T9
15
	D15
	T15
	10
	2
15
	T15
	25
	10
	2
D15
	0
	15
	D10
	D2
T15
	15
	15
	T10
	T2
16
	D16
	T16
	7
	8
16
	T16
	25
	7
	8
D16
	0
	16
	D7
	D8
T16
	16
	16
	T7
	T8
17
	D17
	T17
	2
	3
17
	T17
	25
	2
	3
D17
	0
	17
	D2
	D3
T17
	17
	17
	T2
	T3
18
	D18
	T18
	1
	4
18
	T18
	25
	1
	4
D18
	0
	18
	D1
	D4
T18
	18
	18
	T1
	T4
19
	D19
	T19
	3
	7
19
	T19
	25
	3
	7
D19
	0
	19
	D3
	D7
T19
	19
	19
	T3
	T7
20
	D20
	T20
	5
	1
20
	T20
	25
	5
	1
D20
	0
	20
	D5
	D1
T20
	20
	20
	T5
	T1
25
	1
	2
	3
	4
	5
	6
	7
	8
	9
	10
	11
	12
	13
	14
	15
	16
	17
	18
	19
	20
	D25
D25
	25
